types/resource.go

```diff
@@ -776,3 +778,52 @@ type RecurringJobSpec struct {
 type RecurringJobStatus struct {
 	OwnerID string `json:"ownerID"`
 }
+
+const (
+	SupportBundleLabelKey = "longhorn.io/supportbundle"
+
+	SupportBundleManager = "support-bundle-manager"
+
+	SupportBundleCreationTimeout = 60 * time.Minute
+)
+
+type SuppportBundleState string
+
+const (
+	SupportBundleStateNone             = SuppportBundleState("")
+	SupportBundleStateInProgress       = SuppportBundleState("InProgress")
+	SupportBundleStateReadyForDownload = SuppportBundleState("ReadyForDownload")
+	SupportBundleStateError            = SuppportBundleState("Error")
+)
+
+type SupportBundleError string
+
+const (
+	BundleErrorMkdirFailed = SupportBundleError("Failed to create bundle file directory")
+	BundleErrorZipFailed   = SupportBundleError("Failed to compress the support bundle files")
+	BundleErrorOpenFailed  = SupportBundleError("Failed to open the compressed bundle file")
+	BundleErrorStatFailed  = SupportBundleError("Failed to compute the size of the compressed bundle file")
+
+	SupportBundleProgressPercentageYaml  = 20
+	SupportBundleProgressPercentageLogs  = 75
+	SupportBundleProgressPercentageTotal = 100
+)
+
+type SupportBundleSpec struct {
+	Name               string              `json:"name"`
+	FileName           string              `json:"filename"`
+	State              SuppportBundleState `json:"state"`
```

Comment:
Why `state, size, error, progressPercentage` are in the spec? Those are not user-defined.

Also, how is `fileName` decided? Is it user defined?